### ISSUE TITLE
[BUG] Using `&` or any query param special character in the query is being passed to engines instead of URL encoded

### ISSUE THREAD

**Issue by k4lizen** [hearchco/agent#431]
search query: `convert at&t assembly to intel`
![image](https://github.com/user-attachments/assets/5b5cc76d-953a-4fb5-b186-410620b9f353)
![image](https://github.com/user-attachments/assets/d4cb8fc2-50c3-4975-ba1a-d3049d918bd8)
The results from hearchco are quite bad, also none of the main google results seem to come up.

**Comment by aleksasiriski** [comment]
What about when you drop the &? It looks to me as if this is still the same bug on frontend, adding & removes the rest of the query when encoding (before sending the request to agent)

**Comment by aleksasiriski** [comment]
Actually, after further testing, I think this is because on agent we do not encode queries:

```go
urll := fmt.Sprintf("%v?q=%v%v", searchURL, query, combinedParams)
```

Will try to fix this with proper URL building. This is an oversight from the very beginning of the project :)

**Comment by aleksasiriski** [comment]
@k4lizen Can you confirm if this is fixed on v0.30.0 (running on https://hearch.co)?

**Comment by k4lizen** [comment]
yup all good! gj
